monitoring/consul_controller.go

```diff
@@ -23,60 +23,54 @@ import (
 	"fmt"
 	"github.com/hashicorp/consul/api"
 	"github.com/op/go-logging"
+	"github.com/pkg/errors"
 	"net/http"
 	"os"
 )
 
-type MonitoringController struct {
-	Shutdown         chan bool
-	ShutdownComplete chan bool
-
-	httpAddress string
-	httpPort    int
-	httpServer  *MonitoringServer
-
-	logger *logging.Logger
-
-	promMetrics *PromMetrics
+type ConsulIntegrationController struct {
+	NoIntegrationController
 
 	consulClient    *api.Client
 	consulServiceID string
 }
 
-func NewMonitoringController(address string, port int, consul *api.Client, logger *logging.Logger) (*MonitoringController, error) {
+func NewConsulIntegrationMonitoringController(address string, port int, consul *api.Client, logger *logging.Logger) (*ConsulIntegrationController, error) {
```

Comment:
Maybe have both `NewConsulIntegrationMonitoringController` and `NewNoIntegrationMonitoringController` just return the `Controller` interface? That way, the `ConsulIntegrationController` and `NoIntegrationController` types could remain private.

Reply:
:+1: